version.go

```diff
@@ -104,12 +104,16 @@ func (vs *version) advertiseAndWait() bool {
 
 // serveKey is the entrypoint for incoming HTTP requests.
 func (vs *version) serveKey(w http.ResponseWriter, r *http.Request, key string) {
-	res, err := vs.get(key, r)
+	res, err := vs.get(r, key)
 
 	if err == errNoAvailablePeers {
-		// All of our peers failed us. 502.
-		log.Printf("All peers failed to respond for /%s/%s (version %s)", vs.db, key, vs.name)
+		// Something is wrong with sharding. 502
```

Comment:
It could be something wrong with sharding, but we also error with `errNoAvailablePeers` when every peer errors. I wouldn't necessarily blame that on sharding. I don't know if we want to separate out errors for those two cases or change this description.